pysat/_instrument.py

```diff
@@ -1065,18 +1065,41 @@ def __setitem__(self, key, new_data):
                         # 'Epoch'
                         self.data[key] = (epoch_name, in_data)
                     elif len(in_data) == 1:
-                        # Only provided a single number in iterable, make that
-                        # the input for all times
-                        self.data[key] = (epoch_name,
-                                          [in_data[0]] * len(self.index))
+                        # Only provided a single number in iterable.
+                        if key in self.variables:
+                            # If it already exists, assign as defined.
+                            self.data[key] = in_data
+                        else:
+                            # Otherwise broadcast over time.
+                            warnings.warn(' '.join(('Input for {:}'.format(key),
```

Comment:
Need to update the Warnings section of the docstring.

Reply:
Update the Raises section? Or add Warnings?

Reply:
Raises is only for Errors, add a Warnings section.